prow/jenkins/jenkins.go

```diff
@@ -147,25 +142,42 @@ type BearerTokenAuthConfig struct {
 	Token string
 }
 
-func NewClient(url string, authConfig *AuthConfig, metrics *ClientMetrics) *Client {
+func NewClient(url string, authConfig *AuthConfig, logger *logrus.Entry, metrics *ClientMetrics) *Client {
+	if logger == nil {
+		logger = logrus.NewEntry(logrus.StandardLogger())
+	}
 	return &Client{
-		logger:     logrus.WithField("client", "jenkins"),
+		logger:     logger.WithField("client", "jenkins"),
 		baseURL:    url,
 		authConfig: authConfig,
 		client:     &http.Client{},
 		metrics:    metrics,
 	}
 }
 
-func (c *Client) log(methodName string, args ...interface{}) {
+func (c *Client) log(msg string, fields logrus.Fields, lvl logrus.Level, args ...interface{}) {
```

Comment:
Not super sure of the utility of this method anymore -- if we push as much string-formatting into `WithFields()` calls this becomes less useful, right? Just pick the right function at the callsite.

Reply:
The only issue I see here is the nil check but I don't really mind removing it.